adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/CommandSubscription.java

```diff
@@ -101,14 +102,14 @@ private CommandSubscription(final String topic, final Device authenticatedDevice
             }
         } else {
             isAuthenticated = true;
-            if ((tenant != null && !authenticatedDevice.getTenantId().equals(tenant)) ||
-                    (deviceId != null && !authenticatedDevice.getDeviceId().equals(deviceId))) {
+            if (tenant != null && !authenticatedDevice.getTenantId().equals(tenant)) {
```

Comment:
I think the message in the thrown exception needs to be adapted as well

Reply:
Yes, I've adapted it.